### ISSUE TITLE
iterative_train_test_split should support pd.Dataframe as X (with solution)

### ISSUE THREAD

**Issue by kretes** [scikit-multilearn/scikit-multilearn#206]
currently `iterative_train_test_split` just accepts numpy arrays, both as X and y. It is sometimes useful do the split when X is a pd.Dataframe. 

It would be good to have this support built-in. For now I've created a simple solution translating df to an index array and back again:

`
def iterative_train_test_split_dataframe(X, y, test_size):
    df_index = np.expand_dims(X.index.to_numpy(), axis=1)
    X_train, y_train, X_test, y_test = iterative_train_test_split(df_index, y, test_size = test_sizes.get(s, DEFAULT_SIZE))
    X_train = X.loc[X_train[:,0]]
    X_test = X.loc[X_test[:,0]]
    return X_train, y_train, X_test, y_test
`

and a kind-of-test that it works the same:

```
N_SAMPLES=10
N_X = 4
N_LABELS = 5
X = np.random.rand(N_SAMPLES, N_X)
y = np.random.rand(N_SAMPLES, N_LABELS) > 0.5

np.random.seed(42)
X_train, y_train, X_test, y_test = iterative_train_test_split(X, y, test_size = 0.5)

# dataframe:
X_df = pd.DataFrame(X)

# TypeError:
#X_train, y_train, X_test, y_test = iterative_train_test_split(X_df, y, test_size = 0.5)

np.random.seed(42)
X_train_df, y_train_df, X_test_df, y_test_df = iterative_train_test_split_dataframe(X_df, y, 0.5)

(
(X_train_df.to_numpy() == X_train).all(),
(y_train_df == y_train).all(),
(X_test_df.to_numpy() == X_test).all(),
(y_test_df == y_test).all()    
)

(True, True, True, True)
```

**Comment by ianyu93** [comment]
Actually, has there been a follow up on this issue?

**Comment by erikhuck** [comment]
My understanding is the current state of `iterative_train_test_split` is that it only accepts pandas DataFrames.

**Comment by tomeeeS** [comment]
Would be pretty great if it accepted DataFrames too...
I have my data in df's (who doesn't?!...), and want to separate a test set, save it, and then filter the training set further (based on some logic that's supposed to help me throw away worse samples, but I want to keep all kinds of samples in the test set).
It's cumbersome to have to work around this lack of functionality...

**Comment by tomeeeS** [comment]
Here's my workaround - here both X and y are DataFrames:

```
def iterative_train_test_split_dataframe(X, y, test_size):
    noob_train_test_split = iterative_train_test_split
    X_train, y_train, X_test, y_test = noob_train_test_split(X.reset_index().to_numpy(), y.to_numpy(), test_size=test_size)

    df_X_train = pd.DataFrame(X_train[:, 1:], columns=X.columns, index=X_train[:, 0])
    df_y_train = pd.DataFrame(y_train, columns=y.columns, index=df_X_train.index)
    df_X_test = pd.DataFrame(X_test[:, 1:], columns=X.columns, index=X_test[:, 0])
    df_y_test = pd.DataFrame(y_test, columns=y.columns, index=df_X_test.index)
    return df_X_train, df_y_train, df_X_test, df_y_test
```
(It makes the index be a column temporarily for the X's; and forgive my little jab ;) )
